drivers/include/ph_oem.h

```diff
@@ -0,0 +1,366 @@
+ * @return
+ */
+int ph_oem_init(ph_oem_t *dev, const ph_oem_params_t *params);
+
+/**
+ * @brief   Sets a new address to the pH OEM device by unlocking the
+ *          @ref PH_OEM_REG_UNLOCK register and  writing a new address to
+ *          the @ref PH_OEM_REG_ADDRESS register.
+ *          The device address will also be updated in the device descriptor so
+ *          it is still usable.
+ *
+ *          Settings are retained in the sensor if the power is cut.
+ *
+ *          The address in the device descriptor will reverse to the default
+ *          address you provided thru PH_OEM_PARAM_ADDR after the
+ *          microcontroller restarts
+ *
+ * @param[in] dev   device descriptor
+ * @param[in] addr  new address for the device. Range: 0x01 - 0x7f
+ *
+ * @return @ref PH_OEM_OK on success
+ * @return @ref PH_OEM_WRITE_ERR if writing to the device failed
+ */
+int ph_oem_set_i2c_address(ph_oem_t *dev, uint8_t addr);
+
+/**
+ * @brief   Enable the interrupt gpio pin if @ref ph_oem_params_t.interrupt_pin
+ *          is defined
+ *
+ * @param[in] dev       device descriptor
+ * @param[in] cb        callback called when the pH OEM interrupt pin fires
+ * @param[in] arg       callback argument
+ * @param[in] option    @ref ph_oem_irq_option_t
+ * @param[in] gpio_mode @ref gpio_mode_t <br>
+ *                      most likely @ref GPIO_IN, else probably
+ *                      @ref GPIO_IN_PU for @ref PH_OEM_IRQ_FALLING or <br>
+ *                      @ref GPIO_IN_PD for @ref PH_OEM_IRQ_RISING and
+ *                      @ref PH_OEM_IRQ_BOTH
+ *
+ * @return @ref PH_OEM_OK on success
+ * @return @ref PH_OEM_WRITE_ERR if writing to the device failed
+ * @return @ref PH_OEM_INTERRUPT_GPIO_UNDEF if the interrupt pin is undefined
+ * @return @ref PH_OEM_GPIO_INIT_ERR if initializing the interrupt gpio pin failed
+ */
+int ph_oem_enable_interrupt(ph_oem_t *dev, ph_oem_interrupt_pin_cb_t cb,
+                            void *arg, ph_oem_irq_option_t option,
+                            gpio_mode_t gpio_mode);
+
+/**
+ * @brief   Set the behavior of the interrupt pin of the pH OEM by setting the
+ *          @ref PH_OEM_REG_INTERRUPT register.
+ *          Pin is disabled by default.
+ *
+ *          The pin will not auto reset on option @ref PH_OEM_IRQ_RISING
+ *          and @ref PH_OEM_IRQ_FALLING after interrupt fires,
+ *          so call this function again to reset the pin state, providing the
+ *          same option (@ref PH_OEM_IRQ_RISING or @ref PH_OEM_IRQ_FALLING).
+ *
+ *          Settings are not retained if the power is cut, so you have to call
+ *          this function again after powering on the device.
+ *
+ * @param[in] dev    device descriptor
+ * @param[in] option @ref ph_oem_irq_option_t
+ *
+ * @return @ref PH_OEM_OK on success
+ * @return @ref PH_OEM_WRITE_ERR if writing to the device failed
+ */
+int ph_oem_set_interrupt_pin(const ph_oem_t *dev, ph_oem_irq_option_t option);
```

Comment:
I'm not really convinced by having this function in the public API like this. Though I understand that a function that resets the interrupt pin is needed. Do you think maybe the irq option could be moved to the device descriptor parameters, and just have a `ph_oem_reset_interrupt_pin` function or something like that which would be called from the callback?

Reply:
ph_oem_set_interrupt_pin() is now private. The IRQ option is now defined in the device descriptor and set in ph_oem_params.h as 'PH_OEM_IRQ_BOTH' by default, if not otherwise defined in the makefile. Calling the interrupt is ready to work after calling ph_oem_enable_interrupt. 

Added the ph_oem_reset_interrupt_pin() function. Function will be called in the callback or in the main in case of powering the device off and on again.

Removed the irq option 'PH_OEM_IRQ_DISABLED' because I dont think its really needed. If the interrupt pin is defined, this option has no real use. What do you think?

Reply:
>Removed the irq option `PH_OEM_IRQ_DISABLED `because I dont think its really needed. If the interrupt pin is defined, this option has no real use. What do you think?

Yes, I believe there is no use for the disable